### PR TITLE
New package: intel-parallel-studio-xe-2019.5.281_1

### PR DESCRIPTION
**Pull request by jbenden**
This packages nearly all of the Intel Parallel Studio XE suite. It's commercial software.

I have been using the resultant packages for a bit now, and have worked out all kinks I've ran up against. I feel it's very useful for other folks now.

Signed-off-by: Joseph Benden <joe@benden.us>

### CODE REVIEW

**Comment by xtraeme** [comment]
I'm not sure if this allows redistribution and it's nonfree software so that you'll have to set
`repository=nonfree` and `restricted=yes`

**Comment by jbenden** [comment]
I completely forgot. Yes, it does not permit redistribution. I've updated the template.

I wish for inclusion of this software, due to the absolute pain it is to install without a package (especially the kernel modules).

Thanks!!

File: srcpkgs/intel-parallel-studio-xe/intel-compiler-base.INSTALL [hunk 1]
```diff
@@ -0,0 +1,7 @@
+case "$ACTION" in
+post)
+	if [ -x sbin/ldconfig ]; then
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/intel-compiler-base.INSTALL]: usr/bin

File: srcpkgs/intel-parallel-studio-xe/template [hunk 2]
```diff
@@ -0,0 +1,591 @@
+# Template file for 'intel-parallel-studio-xe'
+_remove_docs=true
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: Make them build options.

File: srcpkgs/intel-parallel-studio-xe/template [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+_remove_static_objects_mkl=false
+_remove_static_objects_ipp=false
+
+_year=2019
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: Derive them from version.

File: srcpkgs/intel-parallel-studio-xe/template [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+_v_a=5
+_v_b=281
+_update=5
+_sp='cluster_edition'
+_icc_ver='19.0.5'
+_vtune_ver='2019.6.0.602217'
+_inspector_ver='2019.5.0.602103'
+_advisor_ver='2019.5.0.602216'
+_dir_nr=15809
+_parallel_studio_xe_dir="parallel_studio_xe${_year:+_${_year}}${_update:+_update${_update}}${_sp:+_${_sp}}"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: Why wouldn't _year and _sp be set?

File: srcpkgs/intel-parallel-studio-xe/template [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+
+pkgname=intel-parallel-studio-xe
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: Please make one block of variables, in order of xlint, with custom ones placed after revision.

File: srcpkgs/intel-parallel-studio-xe/template [hunk 6]
```diff
@@ -0,0 +1,591 @@
...
+version=${_year}.${_v_a}.${_v_b}
+revision=1
+wrksrc="${_parallel_studio_xe_dir}"
+archs="x86_64"
+repository="nonfree"
+short_desc="Intel C++ C and FORTRAN compiler"
+maintainer="Joseph Benden <joe@benden.us>"
+license="custom"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: custom:Proprietary

File: srcpkgs/intel-parallel-studio-xe/template [hunk 7]
```diff
@@ -0,0 +1,591 @@
...
+homepage="https://intel.com/"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: Anything more specific?

File: srcpkgs/intel-parallel-studio-xe/template [hunk 8]
```diff
@@ -0,0 +1,591 @@
...
+distfiles="http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/${_dir_nr}/${_parallel_studio_xe_dir}.tgz"
+checksum=c03421de616bd4e640ed25ce4103ec9c5c85768a940a5cb5bd1e97b45be33904
+subpackages="intel-common-libs intel-openmp intel-compiler-base intel-mpi intel-fortran-compiler intel-ipp intel-mkl intel-tbb-psxe intel-vtune-amplifier intel-vtune-amplifier-dkms intel-advisor intel-inspector"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: variable is only needed if not all packages specified if some subpackages are built conditionally, or order of creation matters, is it case here?

File: srcpkgs/intel-parallel-studio-xe/template [hunk 9]
```diff
@@ -0,0 +1,591 @@
...
+
+repository=nonfree
+restricted=yes
+nocross=yes
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: It's just repack, so cross build from non-x86 should be ok.

File: srcpkgs/intel-parallel-studio-xe/template [hunk 10]
```diff
@@ -0,0 +1,591 @@
...
+nodebug=yes
+nopie=yes
+nostrip=yes
+
+lib32disabled=yes
+
+makedepends="bsdtar sed"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: these are part of base-devel, not need to write them

File: srcpkgs/intel-parallel-studio-xe/template [hunk 11]
```diff
@@ -0,0 +1,591 @@
...
+depends="bash gcc hicolor-icon-theme desktop-file-utils"
+
+if [ "$XBPS_MACHINE" = "x86_64" ]; then
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: XBPS_TARGET_MACHINE, here and there

File: srcpkgs/intel-parallel-studio-xe/template [hunk 12]
```diff
@@ -0,0 +1,591 @@
...
+			if [[ "$XBPS_MACHINE" != 'i686' ]]; then
+				extract=false
+			fi
+		else
+			if [[ "$XBPS_MACHINE" == 'i686' ]]; then
+				extract=false
+			fi
+		fi
+	fi
+	if $extract ; then
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: extract="" or extract=yes  then if [ "$extract" ] please

File: srcpkgs/intel-parallel-studio-xe/template [hunk 13]
```diff
@@ -0,0 +1,591 @@
...
+		_rpm_dir="$3"
+	fi
+	for rpm_file in ${_rpm_dir}/$1 ; do
+		extract_rpm ${rpm_file} $2
+	done
+}
+
+set_build_vars() {
+	_composer_xe_dir="compilers_and_libraries_${_year}.${_v_a}.${_v_b}"
+	rpm_dir=${wrksrc}/rpm
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: prefix all custom vars with underscore

File: srcpkgs/intel-parallel-studio-xe/template [hunk 14]
```diff
@@ -0,0 +1,591 @@
...
+}
+
+do_install() {
+	set_build_vars
+
+	# clean build directories
+	[ -d ${xe_build_dir} ] && rm -fr ${xe_build_dir}
+	mkdir -p ${xe_build_dir}
+
+	echo -e "-----------------------------------------------------------------------------------"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: I doubt anyone will see this. License is installed anyway.

File: srcpkgs/intel-parallel-studio-xe/template [hunk 15]
```diff
@@ -0,0 +1,591 @@
...
+	extract_rpms 'intel-psxe-*.rpm' $xe_build_dir
+
+	vlicense ${wrksrc}/license.txt
+
+	mv $xe_build_dir/opt ${PKGDESTDIR}
+}
+
+intel-common-libs_package() {
+	short_desc="Intel Common Libraries $_icc_ver"
+	shlib_provides="libintlc.so.5 libcilkrts.so.5 libpdbx.so.5 libimf.so libsvml.so libirng.so libchkp.so"
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: is it necesary?

File: srcpkgs/intel-parallel-studio-xe/template [hunk 16]
```diff
@@ -0,0 +1,591 @@
...
+		mv ${xe_build_dir}/opt ${PKGDESTDIR}/
+		mv ${xe_build_dir}/etc ${PKGDESTDIR}/
+
+		ln -s ./${_composer_xe_dir}/linux/tbb/ ${PKGDESTDIR}/opt/intel/tbb
+	}
+}
+
+intel-vtune-amplifier_package() {
+	short_desc="Intel VTune Amplifier $_vtune_ver"
+	depends='desktop-file-utils pangox-compat python glibc libgomp libgcc libstdc++ fontconfig freetype libxcb libxkbcommon-x11 libxkbcommon zlib libX11 libGL glib gtk+3 pango atk cairo gdk-pixbuf libXcomposite libXcursor libXdamage libXext libXfixes libXi libXrender libXtst expat libuuid libXrandr libXScrnSaver alsa-lib dbus-libs at-spi2-atk libcups'
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/template]: wrap at ~80 column

File: srcpkgs/intel-parallel-studio-xe/files/EULA.txt [hunk 17]
```diff
@@ -0,0 +1,274 @@
+IMPORTANT - READ BEFORE COPYING, INSTALLING OR USING.
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/files/EULA.txt]: License should be downloaded with distfiles to not become stale.

File: srcpkgs/intel-parallel-studio-xe/files/Makefile-dkms [hunk 18]
```diff
@@ -0,0 +1,132 @@
+#
```
Comment (Chocimier) [inline srcpkgs/intel-parallel-studio-xe/files/Makefile-dkms]: This preferably will be downloaded with distfiles too.

**Comment by jbenden** [comment]
Wow. :)  Thank you so very much for the excellent reviews! It really helps!

File: srcpkgs/intel-parallel-studio-xe/files/Makefile-dkms (hunk 18 above)
Comment (jbenden) [reply srcpkgs/intel-parallel-studio-xe/files/Makefile-dkms]: This is a hand-made file to appease the DKMS way of building kernel modules.

File: srcpkgs/intel-parallel-studio-xe/files/EULA.txt (hunk 17 above)
Comment (jbenden) [reply srcpkgs/intel-parallel-studio-xe/files/EULA.txt]: This shouldn't have been there. Thanks!

File: srcpkgs/intel-parallel-studio-xe/template (hunk 15 above)
Comment (jbenden) [reply srcpkgs/intel-parallel-studio-xe/template]: It seems so. I am having a lot of problems with auto-detection of shared libraries that are exported verse used by packages. It wouldn't be a concern if the whole thing were a single package, but is a problem with this being many packages in one.

File: srcpkgs/intel-parallel-studio-xe/template (hunk 4 above)
Comment (jbenden) [reply srcpkgs/intel-parallel-studio-xe/template]: The last one might not be, depending on which edition one wants. However, I have professional license and it works with enterprise installer... So it seems a run-time detection is used, not a package software installer difference.

**Comment by jbenden** [comment]
I have pushed a new version. I am prepared to be destroyed! ;)

Thank you so much! Any comments, suggestions, or ideas are most welcomed,
-Joe

**Comment by jbenden** [comment]
I just figured out that a number of Python files were removed (post-install) by `xbps-src` hook scripts. But since these compiled Python files are required, I had to concoct a general solution to the problem.

I added a method of defining extra variables that control which hook scripts are skipped. For example, the one I'm using to skip over post-install hooks:
```
skip_post_install_hooks="02-remove-libtool-archives 02-remove-perl-files 02-remove-python-bytecode-files"
```

**Comment by jbenden** [comment]
I have refactored the DKMS kernel module patches; and have created a [new repository for them](https://github.com/jbenden/intel-vtune-dkms). This will help out other distributions which would like them, too.

**Pull request closed without merging**